src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,132 @@
+package duke;
+
+import java.time.LocalDate;
+import java.util.Arrays;
+
+/**
+ * Parser decodes user given instruction to determine the appropriate action to be taken by Duke.
+ */
+public class Parser {
+    private Command command;
+    private String parameters;
+    private LocalDate date;
+
+    /**
+     * Constructs a Parser object. Attempts to extract the different sections of the specified instruction
+     * into the command, parameters, and date (if any).
+     * 
+     * @param instruction Input from user as instructions to Duke.
+     * @throws InvalidInstructionException If specified instruction is invalid.
+     */
+    public Parser(String instruction) throws InvalidInstructionException {
```

Comment:
The conversion between `String instruction` and `Command command` was well done. As has mentioned above, you can also add more functions into this `parser` to spare your efforts elsewhere.